qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +61,50 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def build_sample_summary(study_id, user_id, visible=None):
```

Comment:
Ok, I see now, can you make the loading of the information dynamic? This looks extremely expensive as it is right now...